interpreter/interpreter_test.go

```diff
@@ -582,3 +582,57 @@ func TestBuiltinFunctionUnichar2int(t *testing.T) {
 		}
 	}
 }
+
+func TestEnums(t *testing.T) {
+
+	NewEnumWithIds := func(elements ...interface{}) *runtime.Enum {
+		ret := runtime.NewEnum()
+		for i := 0; i < len(elements); i += 2 {
+
+			name, _ := elements[i+0].(string)
+			index, _ := elements[i+1].(int)
+			ret.Elements[name] = index
+		}
+		return ret
+	}
+
+	tests := []struct {
+		input    string
+		expected runtime.Object
+	}{
+		{"type enumerated E1 {}; E1", runtime.Errorf("this enum has no elements")},
+		{"type enumerated E2 {red}; E2", runtime.NewEnum("red")},
+		{"type enumerated E3 {red, green, blue}; E3", runtime.NewEnum("red", "green", "blue")},
+		{"type enumerated E4 {red(1,2)}; E4", runtime.Errorf("enum element has unexpected number of arguments 2 for key red")},
```

Comment:
This is actually allowed. Example from the [standard](https://www.etsi.org/deliver/etsi_es/201800_201899/20187301/04.14.01_60/es_20187301v041401p.pdf#page=64):
```
type enumerated MyThirdEnumType {
    Blue(0),
    Yellow(1),
    Green(3),
    Other(2, 4..255)
}
```

Let's create an issue for that, so we don't forget to add this feature later.